### DISCUSSION TITLE
idea中使用Copilot无法登录

### DISCUSSION THREAD

**Discussion by yuxiuliang** [community/community#40882] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

GitHub Copilot

Request signInInitiate failed with message: read ECONNRESET, request id: 5, error code: -32603

IntelliJ IDEA 2022.2 (Ultimate Edition)
Build #IU-222.3345.118, built on July 26, 2022
Runtime version: 17.0.3+7-b469.32 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Windows 10 10.0
GC: G1 Young Generation, G1 Old Generation
Memory: 10240M
Cores: 20
Non-Bundled Plugins:
    com.intellij.zh (222.202)
    com.github.copilot (1.1.36.2164)
    com.alibaba.p3c.smartfox (2.1.1)
    MavenRunHelper (4.23.222.2964.0)
    com.baomidou.plugin.idea.mybatisx (1.5.5)

Kotlin: 222-1.7.10-release-334-IJ3345.118

**Comment by ArnoZWang** [comment]
- 尝试过重装插件吗？最好是先卸载插件，然后重启 IDEA（如有必要可以换成重启电脑），再重装插件。
- 如果正在使用代理软件的话，尝试一下开关代理。

P.S. 请问是之前能正常使用，突然出现问题吗？还是从来没有正常使用过。

**Reply by ArnoZWang** [reply]
- Have you ever tried reinstalling the plugin? It's best to uninstall the plugin, then restart IDEA (restarting your computer if you think necessary), then reinstall the plugin.
- If you are using proxy tool, try to switch on/off the proxy.

P.S. Is it that Copilot was working properly before and suddenly there is this issue? Or it has never worked properly?

**Reply by yuxiuliang** [reply]
昨天可以正常使用，今天就无法使用了

**Reply by ArnoZWang** [reply]
我情况跟你一样，也是今天突然故障。
后来我把所有能重装/重启的东西全部试了一遍，包括重装插件、重启软件、重启代理工具，不知道哪一步有效但最终解决了。
希望能帮到你。

**Reply by yuxiuliang** [reply]
谢谢，我把插件降低版本就可以使用了

**Reply by ghost** [reply]
关闭代理工具之后，登录成功

**Reply by zhimoai** [reply]
我的情况是关闭插件后又重新打开就可以了

**Comment by yuxiuliang** [comment]
> ### Select Topic Area
> Product Feedback
> 
> ### Body
> GitHub Copilot
> 
> Request signInInitiate failed with message: read ECONNRESET, request id: 5, error code: -32603
> 
> IntelliJ IDEA 2022.2 (Ultimate Edition) Build #IU-222.3345.118, built on July 26, 2022 Runtime version: 17.0.3+7-b469.32 amd64 VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o. Windows 10 10.0 GC: G1 Young Generation, G1 Old Generation Memory: 10240M Cores: 20 Non-Bundled Plugins: com.intellij.zh (222.202) com.github.copilot (1.1.36.2164) com.alibaba.p3c.smartfox (2.1.1) MavenRunHelper (4.23.222.2964.0) com.baomidou.plugin.idea.mybatisx (1.5.5)
> 
> Kotlin: 222-1.7.10-release-334-IJ3345.118

最新版idea 2022.3 和Copilot 1.1.37.2188  需要给idea设置http 代理才可以使用

**Reply by 98ZP** [reply]
我使用的是您说的idea 2022.3（pycharm） 和Copilot 1.1.37.2188 ，也设置了http代理，测试连接成功，但是依然显示Request signInInitiate failed with message: read ECONNRESET, request id: 10, error code: -32603 。想问下您还有其他办法吗？

**Comment by Milittle** [comment]
我也遇到了类似的问题， 有大佬知道一下么？

**Comment by shoukailiang** [comment]
我也是

**Comment by congim** [comment]
俺也一样

**Comment by ShenLongDX** [comment]
我也是

**Comment by Czh0618** [comment]
最新版idea 2022.3.2 和Copilot 1.1.37.2188 也设置了http代理，但是页面相应成功之后idea就没反应了，登录不上去

**Reply by Czh0618** [reply]
![image](https://user-images.githubusercontent.com/15818613/222646990-480e3271-5ea0-41ac-9279-d2e01a1d3234.png)

**Reply by xm5646** [reply]
我也是 idea2022.3  copilot 1.2.4.2459

**Reply by wujunbiao1002** [reply]
![image](https://user-images.githubusercontent.com/22448122/228742880-5396ebc2-fe1c-4aac-a7a4-91931ef9d5b0.png)
![image](https://user-images.githubusercontent.com/22448122/228742915-1e320e03-d590-493c-acc9-bff4f9ff5564.png)
一样 的问题页面显示登录成功，idea还是没有登录。
idea2023.1 copilot 1.2.4.2459

**Reply by sunmh207** [reply]
the same problem for me

**Reply by almsIsbest** [reply]
我也是这个问题，有解决的伙伴吗

**Reply by wujunbiao1002** [reply]
检查一下自己的GitHub copilot免费试用2个月有没获取成功，它在领取使用后要进行一个支付操作（不扣钱）才算真的领取成功。
![image](https://user-images.githubusercontent.com/22448122/230825802-f5989b45-d461-4210-9fca-a2f529cb3e51.png)

**Reply by lushangkan** [reply]
打开IDEA前删除AppData\Local\github-copilot\hosts.json

**Reply by adoin** [reply]
所以还是copilot自己的问题。。。擦删完解决了

**Reply by lushangkan** [reply]
新版本已修复，还登录不上去就是网络问题了

**Comment by henry-hub** [comment]
Copilot插件重新禁用启用就好了

**Comment by yycode001** [comment]
试试修改hosts文件

140.82.114.3 github.com
199.232.69.194 github.global.ssl.fastly.net
140.82.114.5 api.github.com

**Reply by yzk656** [reply]
谢谢你，这个方法真的可以。

**Reply by ScienceNoBorders** [reply]
+1 🫡

**Reply by CShisan** [reply]
如果开启代理后导致copilot请求失败的，就把github.global.ssl.fastly.net和api.github.com加入到代理的直连名单就可以了

**Reply by qeekblackstar** [reply]
有效解决

**Reply by TJ12342** [reply]
感谢，成功解决

**Reply by Evander-Liam** [reply]
有用，感谢

**Comment by yuxiuliang** [comment]
IntelliJ IDEA 2023.2 最新版如果出现类似错误：Sign in failed. Reason: Request signInInitiate failed with message: connect ETIMEDOUT 20.205.243.168:443, request id: 6, error code: -32603
需要卸载copilot插件并重新安装

**Comment by tienq910** [comment]
我是配置idea的http代理后才好的

**Comment by 20302010018** [comment]
在idea设置里找到http代理，选择手动代理，使用地址127.0.0.1:7890（Clash用户，根据你使用的代理软件而不同）

**Comment by WhiteLee03** [comment]
idea设置proxy，然后用http，用http，用http！不要用socks，不要用socks，不要用socks！
